packages/firebase_ml/android/src/main/java/io/flutter/plugins/firebaseml/FirebaseMLPlugin.java

```diff
@@ -40,21 +50,86 @@ public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBindin
     channel.setMethodCallHandler(this);
   }
 
+  @RequiresApi(api = Build.VERSION_CODES.N)
   @Override
-  public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
+  public void onMethodCall(@NonNull MethodCall call, @NonNull final Result result) {
+    final FirebaseCustomRemoteModel remoteModel;
+    String modelName;
+
     switch (call.method) {
-      case "downloadRemoteModel":
-        result.success("Testing call. Will be removed shortly");
+      case "FirebaseModelManager#download":
+        assert (call.argument("model") != null);
+        assert (call.argument("conditions") != null);
+
+        modelName = call.argument("model");
+        Map<String, Boolean> conditionsToMap = call.argument("conditions");
+        remoteModel = new FirebaseCustomRemoteModel.Builder(modelName).build();
+        FirebaseModelDownloadConditions.Builder conditionsBuilder =
+            new FirebaseModelDownloadConditions.Builder();
+        if (conditionsToMap.get("requireCharging")) conditionsBuilder.requireCharging();
+        if (conditionsToMap.get("requireDeviceIdle")) conditionsBuilder.requireDeviceIdle();
+        if (conditionsToMap.get("requireWifi")) conditionsBuilder.requireWifi();
+
+        FirebaseModelDownloadConditions conditions = conditionsBuilder.build();
+        FirebaseModelManager.getInstance()
+            .download(remoteModel, conditions)
+            .addOnCompleteListener(
+                new OnCompleteListener<Void>() {
+                  @Override
+                  public void onComplete(@NonNull Task<Void> task) {
+                    result.success(remoteModelToMap(remoteModel));
```

Comment:
Is there any error handling necessary here for when task contains an exception?

Reply:
https://firebase.google.com/docs/ml-kit/android/use-custom-models this documentation suggests there is no failure case.

Reply:
Done